spec/request/service_credential_bindings_spec.rb

```diff
@@ -646,6 +657,38 @@ def check_filtered_bindings(*bindings)
       end
     end
 
+    context "last binding operation is in 'delete in progress' state" do
+      before do
+        app_binding.save_with_attributes_and_new_operation({}, { type: 'delete', state: 'in progress' })
+      end
+
+      it 'returns an error' do
+        api_call.call(admin_headers)
+        expect(last_response).to have_status_code(404)
+        expect(parsed_response['errors']).to include(include({
+                                                               'detail' => 'Deletion of service binding in progress',
+                                                               'title' => 'CF-ResourceNotFound',
+                                                               'code' => 10010,
+                                                             }))
+      end
+    end
+
+    context "last binding operation is in 'delete failed' state" do
+      before do
+        app_binding.save_with_attributes_and_new_operation({}, { type: 'delete', state: 'failed' })
+      end
+
+      it 'returns an error' do
+        api_call.call(admin_headers)
+        expect(last_response).to have_status_code(404)
+        expect(parsed_response['errors']).to include(include({
+                                                               'detail' => 'Deletion of service binding failed',
```

Comment:
Any feedback on that error message ? I followed above schema to structure it but i feel like having a 404 with reason Deletion of service binding failed is a bit confusing.

Reply:
It's consistent. The message is only nice to have (some kind of hint) and might point the caller to the `/v3/service_credential_bindings/:guid` endpoint to retrieve more information.